classification/views/exports/classification_export_filter.py

```diff
@@ -492,6 +504,36 @@ def cms_qs(self) -> QuerySet[ClassificationModification]:
         if allele_id := self.allele:
             cms = cms.filter(classification__allele_info__allele_id=allele_id)
 
+        # Export my data
+        if self.record_filters:
+            internal_lab_filters: List[Q] = []
+            for item in re.split(',|\n|\t', self.record_filters):
+                item = item.strip()
+                if ClinGenAllele.CLINGEN_ALLELE_CODE_PATTERN.match(item):
+                    clingen_allele = get_clingen_allele(item)
+                    internal_lab_filters.append(Q(classification__allele_info__allele__clingen_allele=clingen_allele))
+                elif GENE_SYMBOL_PATTERN.match(item):
+                    if gene_match := classification_gene_symbol_filter(item):
+                        internal_lab_filters.append(classification_gene_symbol_filter(item))
+                elif HGVS_UNCLEANED_PATTERN.match(item):
+                    internal_lab_filters.append(Q(Q(classification__evidence__c_hgvs__value=item)))
+                else:
+                    try:
+                        if vc := VariantCoordinate.from_string(item.strip(), self.genome_build):
+                            variant_coordinate = vc.as_internal_symbolic(self.genome_build)
+                            qs = Variant.objects.all()
+                            allele_data = get_results_from_variant_coordinate(self.genome_build, qs, variant_coordinate)
+                            for i in allele_data:
+                                if allele := i.allele:
+                                    internal_lab_filters.append(Q(classification__allele_info__allele=allele))
+                    except ValueError:
+                        print(f"Failed to parse {item}")
```

Comment:
If something isn't a clingen allele code, gene symbol pattern, hgvs, or variant code - this should throw an error if possible.
If need be could extract this code into a separate method so it can be used for validation as well as generating Qs